### PR TITLE
Automatically create an organizational repo

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  }
+}
+
+# Attach the team to the repository
+resource "github_team_repository" "this" {
+  count = "${length(var.access_teams)}"
+
+  team_id    = "${data.github_team.access_teams.*.id[count.index]}"
+  repository = "${github_repository.this.name}"
+  permission = "pull"
```
Comment (lorengordon) [inline main.tf]: I think the permission needs to be a variable, as `pull` is read-only, and we need to be able to set `push` and `admin` privs also. That probably means that `access_teams` needs to be a list of teams/permissions...

```
access_teams = [
  {
    name = "foo"
    permission = "pull"
  },
  {
    name = "bar"
    permission = "push"
  }
]
```

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,71 @@
+provider "github" {}
+
+locals {
+  pr_teams   = "${length(var.restricted_pr_teams) != 0 ? var.restricted_pr_teams : list("")}"
```
Comment (lorengordon) [inline main.tf]: Make sure you are using terraform 0.11.14... the ternary operator cannot return lists until tf 0.12...

Comment (confusdcodr) [reply main.tf]: Ok, so what is the 0.11.14 equivalent of doing this?

Comment (lorengordon) [reply main.tf]: Not really sure I understand why this is a thing, but maybe just set the default to `[""]`?

Comment (confusdcodr) [reply main.tf]: This has been fixed. I managed to overcomplicate it when I was previously having difficulty with getting terraform to recognize the variable as a list and therefore had to put the variable inside brackets `[]`. Anyway, now it seems to work fine. Also downgraded my version of terraform to ensure it works correctly.

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+# Get information about teams that already exist within the organization
+data "github_team" "access_teams" {
+  count = "${length(var.access_teams)}"
+
+  slug = "${lookup(var.access_teams[count.index], "name")}"
+}
+
+data "github_team" "restricted_pr_teams" {
```
Comment (lorengordon) [inline main.tf]: We don't need these two data lookups anymore, right, for pr and push teams?

**Review by lorengordon — APPROVED** [review]

**Pull request merged**